### PR TITLE
add typings for @fluent/bundle, @fluent/dedent, @fluent/langneg and @fluent/react

### PR DESCRIPTION
**Pull request by blushingpenguin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
Is the listing of authors intentional? @huy-nguyen, @JamesNimlos can you review?

**Comment by blushingpenguin** [comment]
Yes, the packages have been renamed and refactored (the originals are named fluent, fluent-react) and these typings are based on the original typings

**Comment by huy-nguyen** [comment]
It looks like these packages have indeed been namespace-ed under `@fluent`. If all the author did was copying over the definitions from the existing packages I think it should be good to go.

**Comment by andrewbranch** [comment]
I feel like the packages at `fluent-react` etc. should be deleted, but I’m not sure exactly how that should be done... normally when we remove packages, it's because the JS library started shipping its own types. @sandersn any advice?

File: types/fluent__dedent/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for @fluent/dedent 0.1
+// Project: http://projectfluent.org
+// Definitions by: Mark Weaver <https://github.com/blushingpenguin>, Huy Nguyen <https://github.com/huy-nguyen>, James Nimlos <https://github.com/jamesnimlos>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.3
+
+// tslint:disable-next-line:npm-naming
+export default function ftl(strings: TemplateStringsArray, ...values: any[]): string;
```
Comment (andrewbranch) [inline types/fluent__dedent/index.d.ts]: Looks like this shouldn’t be a default export; it’s a UMD wrapping a CommonJS export: https://unpkg.com/browse/@fluent/dedent@0.1.0/index.js

```ts
declare function FluentDedent(/* ... */): string;
export = FluentDedent;
export as namespace FluentDedent;
```

**Review by andrewbranch — CHANGES_REQUESTED** [review]
I noticed most of these libraries have UMD wrappers that aren’t represented here. If people use them as globals, they’d need to add an `export as namespace WhateverTheGlobalNameIs`.

**Comment by sandersn** [comment]
@andrewbranch the right thing is for, eg, fluent-react to be deprecated and point to `@fluent/react`. I *think* the below addition to notNeededPackages might work, although I have not tested it. I haven't looked up asOfVersion, but it needs to be a version that's newer than the currently published `@types/fluent-react` and I think it also has to match a version of `@types/fluent__react`.

```json
        {
            "libraryName": "@types/fluent__react",
            "typingsPackageName": "fluent-react",
            "sourceRepoURL": "https://github.com/DefinitelyTyped/DefinitelyTyped",
            "asOfVersion": "?.?.?"
        },
```

Of course the fluent-react files need to be deleted as well.

File: types/fluent__dedent/index.d.ts (hunk 1 above)
Comment (blushingpenguin) [reply types/fluent__dedent/index.d.ts]: Ok, I see what you mean -- I have fixed the dedent definition and added namespaces to the other modules now.  Also I had originally split the definition files as per the module source which was wrong as the packaged modules are all in one file so I've merged them back into one.
Thanks for the help!

**Pull request closed without merging**